scripts/integration/datadog-agent/compose.yaml

```diff
@@ -17,7 +17,7 @@ services:
     volumes:
     - ../../../tests/data/datadog-agent/conf.yaml:/etc/datadog-agent/conf.d/test.d/conf.yaml
   datadog-trace-agent:
-    image: docker.io/datadog/agent:7.31.0
+    image: docker.io/datadog/agent:${DATADOG_AGENT_VERSION}
```

Comment:
I'm not sure if this is the right change. Should both images be the same version, or is there a reason we had them separate?

Reply:
<grumble grumble> I'm fairly sure @jszwedko and I talked about this when I was adding integration tests for the APM stats to the dd traces tests...

I want to say there was a reason we kept it back. But I can't remember.

However, in a general sense, we elected to keep the traces one to the latest offical agent version, in order help catch changes in behavior that we need to adapt vector to. So I think that would ideally be what we'd do for the agent source too. I want to say it had something to do with the api version we were using by default in the agent source, and maybe something with the Agent's default api endpoint for specific 7.x versions. But its fuzzy and I'm not seeing notes about it in the code.